compiler/rustc_const_eval/src/interpret/intrinsics.rs

```diff
@@ -413,35 +413,45 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                         ),
                     )?;
                 }
-                if intrinsic_name == sym::assert_zero_valid
-                    && !layout.might_permit_raw_init(
-                        self,
-                        InitKind::Zero,
+
+                if intrinsic_name == sym::assert_zero_valid {
+                    let should_panic = !crate::might_permit_raw_init(
```

Comment:
Strictly speaking, this doesn't need to be in a variable, but I figured it's cleaner to split it like this, especially since the call is rather long now. I can inline this back to how it was before.